reference/mongodb/bson/persistable/bsonserialize.xml

```diff
@@ -0,0 +1,236 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- $Revision$ -->
+
+<refentry xml:id="mongodb-bson-persistable.bsonserialize" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
+ <refnamediv>
+  <refname>MongoDB\BSON\Persistable::bsonSerialize</refname>
+  <refpurpose>Provides an array or document to serialize as BSON</refpurpose>
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>abstract</modifier> <modifier>public</modifier> <type class="union"><type>array</type><classname>stdClass</classname><classname>MongoDB\BSON\Document</classname></type><methodname>MongoDB\BSON\Persistable::bsonSerialize</methodname>
+   <void />
+  </methodsynopsis>
+  <para>
+   Called during serialization of the object to BSON. The method must return an
+   <type>array</type>, <classname>stdClass</classname>, or
+   <classname>MongoDB\BSON\Document</classname>.
+  </para>
+  <para>
+   Root documents (e.g. a
+   <interfacename>MongoDB\BSON\Persistable</interfacename> passed to
+   <function>MongoDB\BSON\fromPHP</function>) will always be serialized as a
+   BSON document. For field values, associative arrays and
+   <classname>stdClass</classname> instances will be serialized as a BSON
+   document and sequential arrays (i.e. sequential, numeric indexes starting at
+   <literal>0</literal>) will be serialized as a BSON array.
```

Comment:
This looks like copypasta from `serializable/bsonserialize.xml`. Anything returned by `Persistable::bsonSerialize()` will be serialized as a BSON document. Noting this would also provide you the opportunity to explain _why_ this interface narrows the return type of `bsonSerialize()`, since encoding the ODM class requires us to utilize a document type (hence the exclusion of PackedArray and forcing _all_ PHP arrays to encode as a BSON document).

Reply:
Updated the wording to explain that the return value will always be serialised as a document.